client.ts

```diff
@@ -1,44 +1,67 @@
 import { dial } from "deno";
 import { Connection } from "./connection.ts";
+import { Pool } from "./pool.ts";
 import { Query, QueryConfig, QueryResult } from "./query.ts";
-import { IConnectionParams, ConnectionParams } from "./connection_params.ts";
+import { ConnectionParams, IConnectionParams } from "./connection_params.ts";
 
 export class Client {
-  connection: Connection;
-  connectionParams: ConnectionParams;
+  private _connection: Connection;
+  private _connectionParams: ConnectionParams;
+  release: () => void;
 
-  constructor(config?: IConnectionParams | string) {
-    this.connectionParams = new ConnectionParams(config);
+  constructor(
+    config?: IConnectionParams | string | Connection,
+    release?: () => void
+  ) {
+    if (config instanceof Connection) {
```

Comment:
I'd rather not accept `Connection`. I don't even want to expose it as public API

Reply:
This is for internal only. i.e a pool _needs_ to pass the connection.

Reply:
Is it better to add a `@internal` method (that instantiates and sets _connection)?

I agree it's weird to pass it in the constructor...